src/plugins/data/common/search/aggs/buckets/lib/histogram_calculate_interval.ts

```diff
@@ -124,7 +124,11 @@ export const calculateHistogramInterval = ({
   esTypes,
 }: CalculateHistogramIntervalParams) => {
   const isAuto = isAutoInterval(interval);
-  let calculatedInterval = isAuto ? 0 : parseFloat(interval);
+  let calculatedInterval = isAuto
+    ? 0
+    : typeof interval !== 'number'
+    ? parseFloat(interval)
+    : interval;
```

Comment:
Had to tweak logic here to make TS happy once updating `interval` to be string or number. Now we need to check for the case where it's already a number.

Technically this doesn't change anything at runtime (since passing a number to `parseFloat` will still return the same number), but TS expects `parseFloat` to only accept a number.